library/src/main/java/org/apache/apex/malhar/lib/wal/FileSystemWAL.java

```diff
@@ -442,9 +443,16 @@ private Slice readOrSkip(boolean skip) throws IOException
           inputStream = getInputStream(currentPointer);
         }
 
-        if (inputStream != null && currentPointer.offset <
-            fileSystemWAL.fileContext.getFileStatus(currentOpenPath).getLen()) {
-          int len = inputStream.readInt();
+        int ch1;
+
+        if (inputStream != null && (ch1 = inputStream.read()) != -1) {
```

Comment:
@tweise I don't see a big difference between using readInt() and read(). Both are suitable in that place.
@sandeshh Effective Java, IMO, refers to designing API, so it suggests to prefer returning null or -1 over raising an exception. In the case of readInt() there is no way to return a marker that will indicate an invalid value. Note that in this particular case, you may want to see if there is a library that allows reading length, not int. You may also implement helper method readLenght instead of inlining the logic.